pilot/pkg/networking/core/v1alpha3/tls.go

```diff
@@ -260,3 +249,31 @@ func buildSidecarOutboundTCPTLSFilterChainOpts(node *model.Proxy, env *model.Env
 
 	return out
 }
+
+// Get the most specific matching virtual service for a host. If no virtual service exists, return nil.
+// For example, if virtual services with hosts "*" and "*.com" exist, match "abc.com" to "*.com" instead of "*".
+type virtualServiceForHost func(model.Hostname) *v1alpha3.VirtualService
+
+func makeVirtualServiceForHost(configs []model.Config) virtualServiceForHost {
+	byHostname := make(map[model.Hostname]*v1alpha3.VirtualService)
+	hostnames := make(model.Hostnames, 0)
+	for _, config := range configs {
+		vs := config.Spec.(*v1alpha3.VirtualService)
+		if vs != nil {
+			for _, vsHost := range vs.Hosts {
+				hostname := model.Hostname(vsHost)
+				hostnames = append(hostnames, hostname)
+				byHostname[hostname] = vs
+			}
+		}
+	}
+	sort.Sort(hostnames) // order from most to least specific
```

Comment:
where is the sort function?

Reply:
`model.Hostnames` implements the sorting interface and sorts with most to least specific ordering.

Reply:
https://github.com/istio/istio/blob/master/pilot/pkg/model/service.go#L524